tempodb/backend/azure/azure_helpers.go

```diff
@@ -9,40 +9,31 @@ import (
 	"strings"
 	"time"
 
-	"github.com/grafana/tempo/tempodb/backend/instrumentation"
-
-	"github.com/Azure/azure-pipeline-go/pipeline"
-	blob "github.com/Azure/azure-storage-blob-go/azblob"
-	"github.com/Azure/go-autorest/autorest/adal"
-	"github.com/Azure/go-autorest/autorest/azure"
-	"github.com/Azure/go-autorest/autorest/azure/auth"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/policy"
+	"github.com/Azure/azure-sdk-for-go/sdk/azidentity"
+	"github.com/Azure/azure-sdk-for-go/sdk/storage/azblob"
+	"github.com/Azure/azure-sdk-for-go/sdk/storage/azblob/blob"
+	"github.com/Azure/azure-sdk-for-go/sdk/storage/azblob/container"
 	"github.com/cristalhq/hedgedhttp"
+
+	"github.com/grafana/tempo/tempodb/backend/instrumentation"
 )
 
 const (
 	maxRetries = 1
 )
 
-var defaultAuthFunctions = authFunctions{
-	NewOAuthConfigFunc: adal.NewOAuthConfig,
-	NewServicePrincipalTokenFromFederatedTokenFunc: adal.NewServicePrincipalTokenFromFederatedToken,
-}
-
-type authFunctions struct {
-	NewOAuthConfigFunc                             func(activeDirectoryEndpoint, tenantID string) (*adal.OAuthConfig, error)
-	NewServicePrincipalTokenFromFederatedTokenFunc func(oauthConfig adal.OAuthConfig, clientID string, jwt string, resource string, callbacks ...adal.TokenRefreshCallback) (*adal.ServicePrincipalToken, error)
-}
-
-func GetContainerURL(ctx context.Context, cfg *Config, hedge bool) (blob.ContainerURL, error) {
+func getContainerClient(ctx context.Context, cfg *Config, hedge bool) (container.Client, error) {
 	var err error
-	var p pipeline.Pipeline
 
-	retryOptions := blob.RetryOptions{
-		MaxTries: int32(maxRetries),
-		Policy:   blob.RetryPolicyExponential,
+	retry := policy.RetryOptions{
+		MaxRetries:    int32(maxRetries) - 1,
+		TryTimeout:    1 * time.Minute,
```

Comment:
The below three values were set implicitly via `blob.RetryPolicyExponential` in the old SDK. `blob.RetryPolicyExponential` was a shorthand for setting these defaults:
```go
IfDefault(&o.TryTimeout, 1*time.Minute)
IfDefault(&o.RetryDelay, 4*time.Second)
IfDefault(&o.MaxRetryDelay, 120*time.Second)
```

The new SDK contains no shorthand, so we manually set the same defaults. We might consider making these configuration options in Tempo, but I've deliberately not done that in the interest of diff leanness.

Reply:
agreed with these changes. please add a comment noting where these values came from and why they were chosen.

Reply:
Action taken: added this comment:
>  // The values for TryTimeout, RetryDelay and MaxRetryDelay are inherited from the old Azure SDK
> // (azure-storage-blob-go).
> //
> // See https://github.com/Azure/azure-storage-blob-go/blob/905b628ceb292e8d769ae62fb7cc5c5e949360db/azblob/zc_policy_retry.go#L89.